js/cards/disk2.ts

```diff
@@ -685,7 +719,7 @@ export default class DiskII implements Card<State> {
                 break;
             case LOC.DRIVEWRITEMODE: // 0x0f (Q7H)
                 this.debug('Write Mode');
-                this.q7 = false;
+                this.q7 = true;
                 this.writeMode = true;
```

Comment:
In https://github.com/whscullin/apple2js/tree/whscullin/woz_writing I was in the process of merging these. There's also some changes to the PROM code that were required, and some timing work. That got me within one stray `0` of write working as expected - a leading 0 was getting written out instead of starting in with sync bits right away.

Reply:
I'd forgotten about the `woz_writing` branch, so I had a look. Do you remember why you had to change the PROM? And why switching to the 5th cycle for incoming bits was important?

Anyway, my goal here is to not break anything in my next refactoring which will split the WOZ parts and the Nibble parts into separate classes. It also merges `q7` and `writeMode`, like your branch.

I plan to then work on tests for WOZ writing, especially wrapping around at the end of a track. For these tests, I'm going to use precise cycles in sequencer to make sure that—if everything lines up—it works correctly. For all I know, I will run into the same issues, but I hope that splitting these tests from the CPU will make things easier.

Reply:
I think the PROM was wrong because I transcribed it from either a Sather or Beneath PDF by hand at some point and 0s, 8s and Bs all kind of look the same. And since reading worked right away I didn't notice the write portion. Later comparing it to the flow charts I realized the pattern was off.